internal/services/topic.go

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright Strimzi authors.
+// License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+//
+
+// Package services defines some canary related services
+package services
+
+import (
+	"log"
+	"math"
+	"os"
+	"strconv"
+
+	"github.com/Shopify/sarama"
+	"github.com/strimzi/strimzi-canary/internal/config"
+)
+
+type TopicService interface {
+	Reconcile() (int, map[int32][]int32, bool, error)
+	Close()
+}
+
+type topicService struct {
+	canaryConfig *config.CanaryConfig
+	client       sarama.Client
+	admin        sarama.ClusterAdmin
+}
+
+func NewTopicService(canaryConfig *config.CanaryConfig, client sarama.Client) TopicService {
+	admin, err := sarama.NewClusterAdminFromClient(client)
+	if err != nil {
+		log.Printf("Error creating the Sarama cluster admin: %v", err)
+		panic(err)
+	}
+	ts := topicService{
+		canaryConfig: canaryConfig,
+		client:       client,
+		admin:        admin,
+	}
+	return &ts
+}
+
+func (ts *topicService) Reconcile() (int, map[int32][]int32, bool, error) {
```

Comment:
Some doc on the return tuple would be good. Or perhaps the return value should be a type, rather than a 4-tuple?

Reply:
Maybe just a three fields type, leaving the error out because it's mostly a Golang pattern returning a result and eventually an error in a tuple fashion.